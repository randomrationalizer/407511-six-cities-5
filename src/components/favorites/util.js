export const getCitiesFromOffers = (offers) => {
  const cities = offers.slice().map((offer) => offer.city.name).sort();
  return [...new Set(cities)];
};
