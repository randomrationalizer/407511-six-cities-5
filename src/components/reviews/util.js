const REVIEW_MIN_LENGTH = 50;
const REVIEW_MAX_LENGTH = 300;
const RATING_MIN = 1;
const RATING_MAX = 5;

export const checkCommentValidity = (comment) => {
  return comment.length >= REVIEW_MIN_LENGTH && comment.length <= REVIEW_MAX_LENGTH ? true : false;
};

export const checkRatingValidity = (rating) => {
  return rating >= RATING_MIN && rating <= RATING_MAX ? true : false;
};
