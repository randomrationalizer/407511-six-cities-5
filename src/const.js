export const cities = [
  {
    name: `Paris`,
    coords: {
      latitude: 48.864716,
      longitude: 2.349014
    }
  },
  {
    name: `Cologne`,
    coords: {
      latitude: 50.935173,
      longitude: 6.953101
    }
  },
  {
    name: `Brussels`,
    coords: {
      latitude: 50.8505,
      longitude: 4.3488
    }
  },
  {
    name: `Amsterdam`,
    coords: {
      latitude: 52.38333,
      longitude: 4.9
    }
  },
  {
    name: `Hamburg`,
    coords: {
      latitude: 53.551086,
      longitude: 9.993682
    }
  },
  {
    name: `Dusseldorf`,
    coords: {
      latitude: 51.233334,
      longitude: 6.783333
    }
  }
];

export const OfferType = {
  MAIN: `cities`,
  NEARBY: `near-places`,
  FAVORITES: `favorites`
};

export const MapType = {
  MAIN: `cities`,
  PROPERTY: `property`
};

export const SortType = {
  DEFAULT: `DEFAULT`,
  PRICE_UP: `PRICE_UP`,
  PRICE_DOWN: `PRICE_DOWN`,
  RATING_DOWN: `RATING_DOWN`
};
