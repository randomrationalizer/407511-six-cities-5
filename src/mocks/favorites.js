export const favorites = [`1`, `3`, `4`];
