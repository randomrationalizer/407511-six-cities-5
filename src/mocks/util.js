import configureStore from "redux-mock-store";

export const noop = () => {};

export const getMockStore = (store) => {
  const storeConfig = configureStore();
  return storeConfig(store);
};

export const nope = () => {
  return Promise.resolve({});
};
