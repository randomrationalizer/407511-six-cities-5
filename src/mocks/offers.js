import {getPhoto, generateDescription} from "./util.js";


export const offers = [
  {
    id: `1`,
    title: `Beautiful & luxurious apartment at great location `,
    city: `Amsterdam`,
    coords: {
      latitude: 52.3909553943508,
      longitude: 4.85309666406198
    },
    price: 6000,
    type: `apartment`,
    description: generateDescription(),
    bedrooms: 3,
    guests: 4,
    rating: 4.8,
    photos: [
      {
        src: getPhoto(),
        description: ``
      },
      {
        src: getPhoto(),
        description: ``
      },
      {
        src: getPhoto(),
        description: ``
      }
    ],
    options: [`Wi-Fi`, `Washing machine`, `Towels`],
    owner: {
      name: `Angelina`,
      avatar: getPhoto(),
      isHighRated: true
    },
    isPremial: true
  },

  {
    id: `2`,
    title: `Nice, cozy, warm big bed room`,
    city: `Amsterdam`,
    coords: {
      latitude: 52.369553943508,
      longitude: 4.85309666406198
    },
    price: 20,
    type: `room`,
    description: generateDescription(),
    bedrooms: 1,
    guests: 1,
    rating: 3.4,
    photos: [
      {
        src: getPhoto(),
        description: ``
      },
      {
        src: getPhoto(),
        description: ``
      },
      {
        src: getPhoto(),
        description: ``
      },
      {
        src: getPhoto(),
        description: ``
      },
      {
        src: getPhoto(),
        description: ``
      }
    ],
    options: [`Coffee machine`, `Baby seat`, `Kitchen`, `Dishwasher`],
    owner: {
      name: `Mike`,
      avatar: getPhoto(),
      isHighRated: false
    },
    isPremial: false
  },

  {
    id: `3`,
    title: `Canal View Prinsengracht`,
    city: `Amsterdam`,
    coords: {
      latitude: 52.3909553943508,
      longitude: 4.929309666406198
    },
    price: 200,
    type: `hotel`,
    description: generateDescription(),
    bedrooms: 2,
    guests: 3,
    rating: 4.1,
    photos: [
      {
        src: getPhoto(),
        description: ``
      }
    ],
    options: [`Kitchen`, `Dishwasher`, `Cabel TV`, `Fridge`],
    owner: {
      name: `Bob`,
      avatar: getPhoto(),
      isHighRated: true
    },
    isPremial: false
  },

  {
    id: `4`,
    title: ` Wood and stone place`,
    city: `Amsterdam`,
    coords: {
      latitude: 52.3809553943508,
      longitude: 4.939309666406198
    },
    price: 1000,
    type: `house`,
    description: generateDescription(),
    bedrooms: 6,
    guests: 7,
    rating: 3.7,
    photos: [
      {
        src: getPhoto(),
        description: ``
      },
      {
        src: getPhoto(),
        description: ``
      }
    ],
    options: [`Wi-Fi`, `Washing machine`, `Towels`, `Heating`, `Coffee machine`, `Baby seat`, `Kitchen`, `Dishwasher`, `Cabel TV`, `Fridge`],
    owner: {
      name: `James`,
      avatar: getPhoto(),
      isHighRated: false
    },
    isPremial: false
  }
];
