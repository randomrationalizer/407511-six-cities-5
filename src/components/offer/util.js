import dayjs from "dayjs";
import {OfferType} from "../../const";


export const capitalize = (str) => {
  return str.charAt(0).toUpperCase() + str.slice(1);
};

// Возвращает дату в виде строки в формате: 2019-04-24
export const formatDate = (date) => {
  return dayjs(date).format(`YYYY-MM-DD`);
};

// Возвращает дату в виде строки в формате: January 2020
export const humanizeDate = (date) => {
  return dayjs(date).format(`MMMM YYYY`);
};

// Сравнение для сортировки отзывов в хронологическом порядке
export const sortByDate = (a, b) => {
  return dayjs(a.date) - dayjs(b.date);
};

export const getDescriptionSentences = (text) => {
  let description = text.split(`. `);
  description[description.length - 1] = description[description.length - 1].replace(`.`, ``);

  return description;
};

export const isFavoritesCard = (type) => {
  return type === OfferType.FAVORITES;
};

export const isMainPageCard = (type) => {
  return type === OfferType.MAIN;
};
