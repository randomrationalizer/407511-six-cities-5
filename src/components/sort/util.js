import {SortType} from "../../const";

export const sortTypeToTitle = {
  [SortType.DEFAULT]: `Popular`,
  [SortType.PRICE_UP]: `Price: low to high`,
  [SortType.PRICE_DOWN]: `Price: high to low`,
  [SortType.RATING_DOWN]: `Top rated first`
};
