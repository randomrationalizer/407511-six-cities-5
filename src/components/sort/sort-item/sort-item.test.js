import React from "react";
import renderer from "react-test-renderer";
import SortItem from "./sort-item";
import {noop} from "../../../mocks/util";
import {SortType} from "../../../const";


describe(`SortItem component renders correctly`, () => {
  it(`with active state`, () => {
    const tree = renderer
      .create(
          <SortItem
            isActive={true}
            sortType={SortType.DEFAULT}
            onSortItemClick={noop}
          />
      ).toJSON();

    expect(tree).toMatchSnapshot();
  });

  it(`without active state`, () => {
    const tree = renderer
      .create(
          <SortItem
            isActive={false}
            sortType={SortType.DEFAULT}
            onSortItemClick={noop}
          />
      ).toJSON();

    expect(tree).toMatchSnapshot();
  });
});
