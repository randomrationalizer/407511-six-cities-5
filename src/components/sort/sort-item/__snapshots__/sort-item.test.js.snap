// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SortItem component renders correctly with active state 1`] = `
<li
  className="places__option places__option--active"
  onClick={[Function]}
  onKeyDown={[Function]}
  tabIndex="0"
>
  Popular
</li>
`;

exports[`SortItem component renders correctly without active state 1`] = `
<li
  className="places__option "
  onClick={[Function]}
  onKeyDown={[Function]}
  tabIndex="0"
>
  Popular
</li>
`;
